app/services/app/app-decorators.ts

```diff
@@ -0,0 +1,15 @@
+import { getResource } from '../../util/injector';
+import { AppService } from './app';
+
+export function RunInLoadingMode(): any {
+  return function (target: any, methodName: string, descriptor: PropertyDescriptor) {
+    const originalMethod = descriptor.value;
+    descriptor.value = function (...args: any[]) {
+      const appService = getResource<AppService>('AppService');
```

Comment:
This is cool.  I have wondered before the best way to use a service outside of a class.